contracts/FleekERC721.sol

```diff
@@ -278,6 +267,176 @@ contract FleekERC721 is ERC721, FleekAccessControl {
         emit NewTokenDescription(tokenId, _tokenDescription, msg.sender);
     }
 
+    /**
+     * @dev Add a new AccessPoint register for an app token.
+     * The AP name should be a DNS or ENS url and it should be unique.
+     * Anyone can add an AP but it should requires a payment.
+     *
+     * May emit a {NewAccessPoint} event.
+     *
+     * Requirements:
+     *
+     * - the tokenId must be minted and valid.
+     *
+     * IMPORTANT: The payment is not set yet
+     */
+    function addAccessPoint(uint256 tokenId, string memory apName, string memory source) public payable {
```

Comment:
I'm thinking maybe we remove source? If we have tokenId here then the source really comes from that. The source in the UI is to allow someone to create an AP even if an NFA is not created but that we should probably add a new function for.

Reply:
And this can be post MVP and depends on what we design for UI too.

Reply:
Source removed